command/monitor_test.go

```diff
@@ -0,0 +1,102 @@
+package command
+
+import (
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/vault/helper/testhelpers"
+	"github.com/mitchellh/cli"
+)
+
+func testMonitorCommand(tb testing.TB) (*cli.MockUi, *MonitorCommand) {
+	tb.Helper()
+
+	ui := cli.NewMockUi()
+	return ui, &MonitorCommand{
+		BaseCommand: &BaseCommand{
+			UI: ui,
+		},
+	}
+}
+
+func TestMonitorCommand_Run(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name string
+		args []string
+		out  string
+		code int
+	}{
+		{
+			"valid",
+			[]string{
+				"-log-level=debug",
+			},
+			"",
+			0,
+		},
+		{
+			"too_many_args",
+			[]string{
+				"-log-level=debug",
+				"foo",
+			},
+			"Too many arguments",
+			1,
+		},
+		{
+			"unknown_log_level",
+			[]string{
+				"-log-level=haha",
+			},
+			"haha is an unknown log level",
+			1,
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			client, closer := testVaultServer(t)
+			defer closer()
+
+			ui, cmd := testMonitorCommand(t)
+			cmd.client = client
+
+			var code int
+			stopCh := make(chan struct{})
+
+			testhelpers.GenerateDebugLogs(t, stopCh, client)
+
+			go func() {
+				code = cmd.Run(tc.args)
+			}()
+
+			select {
+			case <-time.After(3 * time.Second):
+				close(stopCh)
+				syscall.Kill(syscall.Getpid(), syscall.SIGTERM)
```

Comment:
Seems a bit heavy, why not just t.Fatal?

Reply:
My thought process was to try and kill it the way it would be killed by a user, in order to check the exit code below.  I'll try `t.Fatal` though and see if that works just as well.

Reply:
My concern is that it looks we'll be killing ourself, i.e. the test.

Reply:
If the intent is to stop `cmd.Run`, you can send an empty struct to `cmd.ShutdownCh` like so to shut it down:
```
cmd.ShutdownCh <- struct{}{}
```

You'll have to update `testMonitorCommand` to instantiate that field
```
return ui, &MonitorCommand{
		BaseCommand: &BaseCommand{
			UI: ui,
		},
		ShutdownCh: MakeShutdownCh(),
	}
...
```

Reply:
Ah, very cool.  I'll give that a shot.

Reply:
It looks like `ShutdownCh` is only implemented by a few commands?

Reply:
Ah, right. It might make sense to implement it here it's this is a polling command.

Reply:
Yeah, good call.  I took your suggestion and I think the result is better than before.  Thanks!